operator/cmd/mesh/manifest-apply.go

```diff
@@ -139,6 +141,22 @@ func InstallManifests(setOverlay []string, inFilenames []string, force bool, dry
 	if err != nil {
 		return err
 	}
+
+	serverVersion, err := clientset.Discovery().ServerVersion()
```

Comment:
@jwendell  I'm thinking if this warning message can be moved as a function in `pkg/install/k8sversion/version.go`? 
That might be useful at other places where incompatible version warning is required.

Reply:
I think this is valid, but in a follow-up, for example, while dealing with https://github.com/istio/istio/issues/22740. I'm trying to keep this PR minimum for a backport to 1.7.